src/Site.js

```diff
@@ -892,6 +893,17 @@ class Site {
 
       // eslint-disable-next-line global-require, import/no-dynamic-require
       this.plugins[plugin] = require(pluginPath || plugin);
+
+      if (!this.plugins[plugin].getLinks && !this.plugins[plugin].getScripts) {
+        return;
+      }
+
+      // For resolving plugin asset source paths later
+      this.plugins[plugin]._pluginAbsolutePath = path.dirname(require.resolve(pluginPath || plugin));
+      this.plugins[plugin]._pluginAssetOutputPath = path.resolve(this.outputPath,
+                                                                 PLUGIN_SITE_ASSET_FOLDER_NAME, plugin);
+
+      fs.mkdirSync(this.plugins[plugin]._pluginAssetOutputPath, { recursive: true });
```

Comment:
Even if plugins have assets, the sources might be absolute right? In that case, there might not be any assets copied to the output folder. Should we defer the creation of the folder until the point that we are copying it over?